pre_commit/clientlib.py

```diff
@@ -112,6 +113,32 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+# should inherit from cfgv.Conditional if sha support is dropped
+class CheckBadRev(cfgv.ConditionalOptional):
+    def check(self, dct: Dict[str, Any]) -> None:
+        super().check(dct)
+
+        if self.key in dct.keys():
+            bad_revs = {
+                'stable',
+                'master',
+                'HEAD',
+            }
+            rev = dct[self.key]
+
+            if '.' not in rev and (
+                not re.match(r'^[a-z0-9]+$', rev) or
```

Comment:
upper case shas are valid git revisions, this should be `a-f0-9A-F0-9`

Reply:
good point! i forgot that letters are also restricted to `a-f` …